Makefile

```
@@ -74,51 +75,61 @@ __check_defined = \
 
+# Include project name in image name so that image name
+# does not conflict with other images during local development
+E2E_IMAGE_NAME := $(PROJECT_ROOT)-$(APP_NAME)-e2e
+
 e2e-build: ## Build the e2e Docker image, if not already built, using ./e2e/Dockerfile
-	docker build -t playwright-e2e -f ./e2e/Dockerfile .
+	docker build -t $(E2E_IMAGE_NAME) -f ./e2e/Dockerfile .
 
-e2e-clean-report: ## Remove the local ./e2e/playwright-report and ./e2e/test-results folder and their contents
+e2e-clean-report: ## Remove the local e2e report folders and content
 	rm -rf ./e2e/playwright-report
 	rm -rf ./e2e/blob-report
 	rm -rf ./e2e/test-results
+	rm -rf ./e2e/example-screenshot.png
 
 e2e-delete-image: ## Delete the Docker image for e2e tests
-	@docker rmi -f playwright-e2e 2>/dev/null || echo "Docker image playwright-e2e does not exist, skipping."
+	@docker rmi -f $(E2E_IMAGE_NAME) 2>/dev/null || echo "Docker image $(E2E_IMAGE_NAME) does not exist, skipping."
 
-e2e-merge-reports: ## Merge Playwright blob reports from multiple shards into an HTML report
-	@cd e2e && npx playwright merge-reports --reporter html blob-report
+e2e-merge-reports: ## Merge E2E blob reports from multiple shards into an HTML report
+	@cd e2e && npm run e2e-merge-reports
 
 e2e-setup-ci: ## Setup end-to-end tests for CI
-	@cd e2e && npm ci
-	@cd e2e && npx playwright install --with-deps
+	@cd e2e && npm run e2e-setup
 
 e2e-setup-native: ## Setup end-to-end tests
 	@cd e2e && npm install
-	@cd e2e && npx playwright install --with-deps
+	@cd e2e && npm run e2e-setup
 
-e2e-show-report: ## Show the ./e2e/playwright-report
-	@cd e2e && npx playwright show-report
+e2e-show-report: ## Show the E2E report
+	@cd e2e && npm run e2e-show-report
 
-e2e-test: ## Run E2E Playwright tests in a Docker container and copy the report locally
+e2e-test: ## Run E2E tests in a Docker container and copy the report locally
 e2e-test: e2e-build
 	@:$(call check_defined, APP_NAME, You must pass in a specific APP_NAME)
 	@:$(call check_defined, BASE_URL, You must pass in a BASE_URL)
 	docker run --rm\
-		--name playwright-e2e-container \
+		--name $(E2E_IMAGE_NAME)-container \
 		-e APP_NAME=$(APP_NAME) \
 		-e BASE_URL=$(BASE_URL) \
 		-e CURRENT_SHARD=$(CURRENT_SHARD) \
 		-e TOTAL_SHARDS=$(TOTAL_SHARDS) \
 		-e CI=$(CI) \
 		-v $(PWD)/e2e/playwright-report:/e2e/playwright-report \
 		-v $(PWD)/e2e/blob-report:/e2e/blob-report \
-		playwright-e2e
+		$(E2E_IMAGE_NAME)
 
-e2e-test-native: ## Run end-to-end tests
+e2e-test-native: ## Run end-to-end tests natively
 	@:$(call check_defined, APP_NAME, You must pass in a specific APP_NAME)
 	@:$(call check_defined, BASE_URL, You must pass in a BASE_URL)
 	@echo "Running e2e tests with CI=${CI}, APP_NAME=${APP_NAME}, BASE_URL=${BASE_URL}"
-	@cd e2e/$(APP_NAME) && APP_NAME=$(APP_NAME) BASE_URL=$(BASE_URL) npx playwright test $(E2E_ARGS)
+	@cd e2e/$(APP_NAME) && APP_NAME=$(APP_NAME) BASE_URL=$(BASE_URL) npm run e2e-test $(E2E_ARGS)
```

nit: we only need to cd into `e2e` rather than `e2e/$(APP_NAME)` right?

```suggestion
	@cd e2e && APP_NAME=$(APP_NAME) BASE_URL=$(BASE_URL) npm run e2e-test $(E2E_ARGS)
```

---

Updated